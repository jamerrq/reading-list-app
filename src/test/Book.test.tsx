import { render } from '@testing-library/react'
import Book from '../app/components/book'
import { test, expect } from 'vitest'
import data from '../app/data/books.json'

const book = data.library[0].book
test('Book', () => {
  const { container } = render(<Book book={book} />)
  expect(container).toMatchSnapshot()
})
