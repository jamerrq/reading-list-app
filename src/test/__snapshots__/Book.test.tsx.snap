// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Book 1`] = `
<div>
  <div
    class="flex flex-col items-center gap-3 p-2 shadow-md ease-in-out hover:scale-105 hover:ease-in-out transition-all duration-300 rounded-md border-2 relative [&>button]:hidden [&>button]:hover:inline [&>a]:hidden [&>a]:hover:inline [&>img]:hover:brightness-50 cursor-pointer select-none group"
    style="box-shadow: 0 0 10px 0 #00000088; border: 2px solid #00000088;"
  >
    <button
      class="group inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-unit-2 !gap-unit-0 data-[pressed=true]:scale-[0.97] transition-transform-colors motion-reduce:transition-none bg-default min-w-unit-10 w-unit-10 h-unit-10 absolute top-[15%] text-white px-2 rounded z-10"
      style="background-color: rgb(59, 130, 246);"
      type="button"
    >
      <svg
        class="tabler-icon tabler-icon-bookmarks"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 7a2 2 0 0 1 2 2v12l-5 -3l-5 3v-12a2 2 0 0 1 2 -2h6z"
        />
        <path
          d="M9.265 4a2 2 0 0 1 1.735 -1h6a2 2 0 0 1 2 2v12l-1 -.6"
        />
      </svg>
    </button>
    <a
      class="absolute top-[45%]"
      href="/book/978-0618640157"
    >
      <button
        class="group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-unit-2 !gap-unit-0 data-[pressed=true]:scale-[0.97] transition-transform-colors motion-reduce:transition-none min-w-unit-10 w-unit-10 h-unit-10 bg-blue-500 text-white px-2 rounded z-10"
        type="button"
      >
        <svg
          class="tabler-icon tabler-icon-info-square-rounded-filled"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2l.642 .005l.616 .017l.299 .013l.579 .034l.553 .046c4.687 .455 6.65 2.333 7.166 6.906l.03 .29l.046 .553l.041 .727l.006 .15l.017 .617l.005 .642l-.005 .642l-.017 .616l-.013 .299l-.034 .579l-.046 .553c-.455 4.687 -2.333 6.65 -6.906 7.166l-.29 .03l-.553 .046l-.727 .041l-.15 .006l-.617 .017l-.642 .005l-.642 -.005l-.616 -.017l-.299 -.013l-.579 -.034l-.553 -.046c-4.687 -.455 -6.65 -2.333 -7.166 -6.906l-.03 -.29l-.046 -.553l-.041 -.727l-.006 -.15l-.017 -.617l-.004 -.318v-.648l.004 -.318l.017 -.616l.013 -.299l.034 -.579l.046 -.553c.455 -4.687 2.333 -6.65 6.906 -7.166l.29 -.03l.553 -.046l.727 -.041l.15 -.006l.617 -.017c.21 -.003 .424 -.005 .642 -.005zm0 9h-1l-.117 .007a1 1 0 0 0 0 1.986l.117 .007v3l.007 .117a1 1 0 0 0 .876 .876l.117 .007h1l.117 -.007a1 1 0 0 0 .876 -.876l.007 -.117l-.007 -.117a1 1 0 0 0 -.764 -.857l-.112 -.02l-.117 -.006v-3l-.007 -.117a1 1 0 0 0 -.876 -.876l-.117 -.007zm.01 -3l-.127 .007a1 1 0 0 0 0 1.986l.117 .007l.127 -.007a1 1 0 0 0 0 -1.986l-.117 -.007z"
            fill="currentColor"
            stroke-width="0"
          />
        </svg>
      </button>
    </a>
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height="300"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fimages-na.ssl-images-amazon.com%2Fimages%2FS%2Fcompressed.photo.goodreads.com%2Fbooks%2F1566425108i%2F33.jpg&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fimages-na.ssl-images-amazon.com%2Fimages%2FS%2Fcompressed.photo.goodreads.com%2Fbooks%2F1566425108i%2F33.jpg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fimages-na.ssl-images-amazon.com%2Fimages%2FS%2Fcompressed.photo.goodreads.com%2Fbooks%2F1566425108i%2F33.jpg&w=256&q=75 2x"
      style="color: transparent; object-fit: cover; border-radius: 5px;"
      width="100"
    />
    <span
      class="text-sm font-semibold"
    >
      El Se√±or de los Anillos
    </span>
  </div>
</div>
`;
